src/stores/roles/actions.ts

```diff
@@ -0,0 +1,37 @@
+import { doRequest, get, post, put } from '@/services/http'
+import { getRoles } from './getters'
+import type { Role } from '@/domain/Role'
+import { roles } from './state'
+import type { Knowledge } from '@/domain/Knowledge'
+import type { Answer } from '@/domain/Answer'
+import type { QuizAnswers } from '@/domain/QuizAnswer'
+
+function setRoles(newRoles: Role[]): void {
+  roles.value = newRoles
+}
+
+async function fetchRoles(): Promise<void> {
+  if (getRoles?.value?.length > 0) {
+    return
+  }
+
+  const fetchedRoles = await get<Role[]>('/roles')
+
+  setRoles(fetchedRoles)
+}
+
+async function getQuizResult(quizAnswer: any) {
+  if (quizAnswer) {
```

Comment:
You just have to check here, because you defined quizAnswer as any. You can get rid of this if